### PR TITLE
Add a bash style-guide readme

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
@mirzak did you forget about this one?

**Comment by mirzak** [comment]
> @mirzak did you forget about this one?

Not forget, but has been pushed down my list.

Before I do an deeper review, I have a question regarding `shell-session`. This tag was new to me. Where does this come from? Is it Grav that supports this? What is the difference between this and e.g `bash`? Reason I have so many questions is that we do not have any occurrences of `shell-session` in the current docs while the style guide would say that this is the prefererad way :)

**Comment by oleorhagen** [comment]
> > @mirzak did you forget about this one?
> 
> Not forget, but has been pushed down my list.
> 
> Before I do an deeper review, I have a question regarding `shell-session`. This tag was new to me. Where does this come from? Is it Grav that supports this? What is the difference between this and e.g `bash`? Reason I have so many questions is that we do not have any occurrences of `shell-session` in the current docs while the style guide would say that this is the prefererad way :)

Yes, if you agree to the `shell-session` markdown, then I will go ahead, and change them pretty much throughout the documentation.

The reason being:
`bash` highlighting is really meant for `bash-scripts`, and not for shell commands, like we are mostly doing. i.e., run these commands in your terminal. If you highlight these with `bash`, then it will highlight bash keywords, and I think it looks a little odd.

If you highlight it with `shell-session`, it knows that this is an interactive shell session, and highlights it accordingly.

See: http://13.53.78.221:8090/foo/getting-started/quickstart-with-raspberry-pi for an example.

**Comment by mirzak** [comment]
Even though I agree with the reasoning, I actually prefer the way "with highlight bash" looks. It just looks more familiar and based on a quick look around it seems the "common" way (looked at Docker, Kubernetes and jfrog docs)

But lets get some more input, tagging stakeholder.

@michaelatmender (UX)
@estenberg

**Comment by oleorhagen** [comment]
> Even though I agree with the reasoning, I actually prefer the way "with highlight bash" looks. It just looks more familiar and based on a quick look around it seems the "common" way (looked at Docker, Kubernetes and jfrog docs)
> 
> But lets get some more input, tagging stakeholder.
> 
> @michaelatmender (UX)
> @estenberg

Yes, sounds good. Note that the bash highlighting is a little bad, and only does the most common keywords. Thus, 

```bash
echo foobar
```

Highlights echo. Whilst:

```bash
ls -la .
```

Does not highlight `ls`

**Comment by mirzak** [comment]
Note that there are differences in github and Grav highlights `bash`. What I like in Grav is that Variables and strings are highlighted when using `bash`. 

Example from: https://docs.mender.io/development/get-started/deploy-a-system-update

**Comment by estenberg** [comment]
Personally I prefer the way bash highlighting is in Grav. Scrolling through these snippets, the coloring makes it easier to read (e.g. string, command like export, variable, etc. https://docs.mender.io/development/get-started/deploy-a-system-update

Also a plus that other sites like Docker docs does the same way.

So my vote would be to keep it as-is (bash).

**Comment by oleorhagen** [comment]
Alright, majority rules. Changing it.

**Comment by oleorhagen** [comment]
Alright, @mirzak demolish me with that final review

File: README-bash-styleguide.markdown [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+foobar
+```
+
+
+### Variables
+
+Variables should always use the `${VARIABLE}` syntax. That is:
+
+```bash
+echo ${VAR}
```
Comment (mirzak) [inline README-bash-styleguide.markdown:76]: We might as well require:

```bash
echo "${VAR}"
```

A better example is:

```bash
cd "${VAR}"
```

We do use plenty of variables for paths, and this should safe-guard us from spaces in paths etc.

File: README-bash-styleguide.markdown [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+#### CLI commands
+
+Code blocks with shell commands should always have the `bash` tag, for syntax highlighting.
+
+i.e.
+
+```bash
+echo foobar
+```
```
Comment (mirzak) [inline README-bash-styleguide.markdown:18]: When viewing this document on e.g github which renders the Markdown, the actual examples on how to do it are hidden and you just see how it should be rendered. (this applies to all examples)

I would assume most people will read this on github? Or?  What do you think?

File: README-bash-styleguide.markdown [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+echo ${VAR}
+```
+
+Not
+
+```bash
+echo $VAR
+```
```
Comment (mirzak) [inline README-bash-styleguide.markdown:83]: Would add something about multi-line vs long lines (that is split long lines in multiple line expressions :))

File: README-bash-styleguide.markdown (hunk 2 above)
Comment (mirzak) [reply README-bash-styleguide.markdown:18]: To clarify,

if you go [here](https://github.com/mendersoftware/mender-docs/blob/127fc3f326acd83a47b9cae3e15bc76c8343f53b/README-bash-styleguide.markdown).

You see:

```bash
echo foobar
```

and not:

    ```bash
    echo foobar
    ```

Comment (oleorhagen) [reply README-bash-styleguide.markdown:18]: ahh, haha. Missed it a whole two times in a row

File: README-bash-styleguide.markdown [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+### Highlighting
+
+#### CLI commands
+
+Code blocks with shell commands should always have the `bash` tag, for syntax highlighting.
+
+i.e.
+
+```````
+
```
Comment (mirzak) [inline README-bash-styleguide.markdown:17]: Remove extra line here. There are several throughout the document, same comment applies to them.

Comment (oleorhagen) [reply README-bash-styleguide.markdown]: What other newlines are you referring to? I only see this one. Are you thinking of between paragraphs?

**Pull request merged**